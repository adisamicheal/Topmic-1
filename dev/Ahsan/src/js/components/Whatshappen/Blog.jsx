import React from 'react';
import _ from 'lodash';
import WhatsHappen from './whatsHappen';
import data from '../../Data';

class Blog extends React.Component {
  constructor() {
    super();

    this.state = {
      selectedView: [data[0], data[1]],
      firstIndexofSelectedView: 0,
    }
    this.next=this.next.bind(this);
    this.previous=this.previous.bind(this);
  }

  next(){
    var {selectedView, firstIndexofSelectedView} = this.state;
    if (firstIndexofSelectedView < data.length - 2) {
      var newview = [...selectedView];

      newview.splice(0,2,data[firstIndexofSelectedView+1],data[firstIndexofSelectedView+2]);
      var newindex = firstIndexofSelectedView +1;
      this.setState({
      selectedView:newview,
      firstIndexofSelectedView:newindex
    })

   }

  }


  previous() {
    var {selectedView, firstIndexofSelectedView} = this.state;
    if(firstIndexofSelectedView > 0 && firstIndexofSelectedView < 5) {
      var newview = [...selectedView];
      newview.splice(0,2,data[firstIndexofSelectedView-1],data[firstIndexofSelectedView])
      var newindex = firstIndexofSelectedView-1;
      
      this.setState({
      selectedView:newview,
      firstIndexofSelectedView:newindex
      })

   }
  }



  render() {
    var {selectedView} = this.state;
    selectedView = selectedView.map((item, i) => {
      return <WhatsHappen key={i} blog={item} next={this.next} previous={this.previous} selectedView={selectedView}/>
    });
  

    return (
    	 <div>
         <div id="sec-2">
              <div className="wraper"> 
                  <div className="des">
                      <h2>What's been happening</h2>
                      <span>What it's like to be a part of our school community. </span>
                  </div>
                  
                  <div className="arrows" onClick={this.previous}>
                      <div className="arrow previous">
                  
                      </div>
                  </div>


  		            {selectedView}

                   <div className="arrows" onClick={this.next}>
                      <div className="arrow next">
                  
                      </div>
                  </div>


              </div>
        </div>
		  </div>
    );
  }
};

export default Blog;