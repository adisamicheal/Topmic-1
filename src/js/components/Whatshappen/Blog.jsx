import React from 'react';
import { number, func, array, shape } from 'prop-types';
import { connect } from 'react-redux';
import WhatsHappen from './whatshappen';
import {
  nextViewToSaga,
  previousViewToSaga,
  initialBlogViewToSaga
} from '../../actions';

export class Blog extends React.Component {
  constructor() {
    super();
    this.next = this.next.bind(this);
    this.previous = this.previous.bind(this);
  }

  componentDidMount() {
    this.props.initialBlogViewToSaga();
  }

  next() {
    this.props.nextViewToSaga();
  }
  previous() {
    this.props.previousViewToSaga();
  }
  createNormalView() {
    const { viewData } = this.props;

    const view = viewData.selectedView.map((item, index) => (
      <WhatsHappen blog={item} key={Math.random()} count={index} />
    ));
    return view;
  }
  createRespView() {
    const { viewData } = this.props;

    const view = viewData.respView.map(item => (
      <WhatsHappen blog={item} key={Math.random()} count={'resp'} />
    ));
    return view;
  }
  render() {
    const view = this.createNormalView();
    const respView = this.createRespView();
    return (
      <div id="sec-2">
        <div className="wraper">
          <div className="des">
            <h2>Whats been happening</h2>
            <span>What its like to be a part of our school community.</span>
          </div>

          <div className="slide-show">
            <div className="arrows left">
              <div
                id="previous"
                className="arrow previous"
                onClick={this.previous}
                onKeyPress={() => {}}
                role="button"
                tabIndex="0"
              />
            </div>

            <div className="arrows right">
              <div
                id="next"
                className="arrow next"
                onClick={this.next}
                onKeyPress={() => {}}
                role="button"
                tabIndex="0"
              />
            </div>

            <div className="whatsHappen-card">
              <div className="wh-normal-view">{view}</div>
              <div className="wh-responsive-view">{respView}</div>
            </div>
          </div>

          <div className="clear" />
        </div>
      </div>
    );
  }
}

Blog.propTypes = {
  nextViewToSaga: func,
  previousViewToSaga: func,
  initialBlogViewToSaga: func,
  viewData: shape({
    // eslint-disable-next-line react/forbid-prop-types
    selectedView: array,
    firstIndexofSelectedView: number
  })
};
Blog.defaultProps = {
  nextViewToSaga: null,
  previousViewToSaga: null,
  initialBlogViewToSaga: null,
  viewData: null
};
function mapStateToProps({ viewData }) {
  return {
    viewData
  };
}

export default connect(mapStateToProps, {
  nextViewToSaga,
  previousViewToSaga,
  initialBlogViewToSaga
})(Blog);
