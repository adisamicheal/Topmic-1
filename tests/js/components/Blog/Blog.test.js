import React from 'react';
import { Blog } from '../../../../src/js/components/Whatshappen/Blog.jsx';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({adapter: new Adapter()});


describe('Tests the Whatshappen Blog Component',()=>{

		let nextViewToSaga = jest.fn();
  		let previousViewToSaga = jest.fn();
  		let initialBlogViewToSaga = jest.fn();

  		let data = {
			selectedView: [{},{}],
			respView: [{}],
			firstIndexofSelectedView: 0,
			indexofRespView: 0
		};

		let wrapper = shallow(<Blog
  			previousViewToSaga = {previousViewToSaga}
  			nextViewToSaga = {nextViewToSaga}
  			initialBlogViewToSaga = {initialBlogViewToSaga}
  			viewData = {data}
			/>);
		
		it('renders the Whatshappen Blog Component',()=>{
			expect(wrapper).toMatchSnapshot();
		});

		it('test that initialBlogViewToSaga fun run onpage load',()=>{

			expect(initialBlogViewToSaga.mock.calls.length).toBe(1);
		});

		it('test that nextViewToSaga fun run',()=>{
				wrapper.find('#next').simulate('click');
				expect(nextViewToSaga.mock.calls.length).toBe(1);
		});

			
		it('test that previousViewToSaga fun run',()=>{
				wrapper.find('#previous').simulate('click');
				expect(previousViewToSaga.mock.calls.length).toBe(1);
		});


		
		

});
