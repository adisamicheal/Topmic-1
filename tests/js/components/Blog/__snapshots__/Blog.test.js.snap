// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Whatshappen Blog Component renders the Whatshappen Blog Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Blog
    initialBlogViewToSaga={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    nextViewToSaga={[MockFunction]}
    previousViewToSaga={[MockFunction]}
    viewData={
      Object {
        "firstIndexofSelectedView": 0,
        "indexofRespView": 0,
        "respView": Array [
          Object {},
        ],
        "selectedView": Array [
          Object {},
          Object {},
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="wraper"
      >
        <div
          className="des"
        >
          <h2>
            Whats been happening
          </h2>
          <span>
            What its like to be a part of our school community.
          </span>
        </div>
        <div
          className="slide-show"
        >
          <div
            className="arrows left"
          >
            <div
              className="arrow previous"
              id="previous"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            />
          </div>
          <div
            className="arrows right"
          >
            <div
              className="arrow next"
              id="next"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            />
          </div>
          <div
            className="whatsHappen-card"
          >
            <div
              className="wh-normal-view"
            >
              <WhatsHappen
                blog={Object {}}
                count={0}
              />
              <WhatsHappen
                blog={Object {}}
                count={1}
              />
            </div>
            <div
              className="wh-responsive-view"
            >
              <WhatsHappen
                blog={Object {}}
                count="resp"
              />
            </div>
          </div>
        </div>
        <div
          className="clear"
        />
      </div>,
      "id": "sec-2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="des"
          >
            <h2>
              Whats been happening
            </h2>
            <span>
              What its like to be a part of our school community.
            </span>
          </div>,
          <div
            className="slide-show"
          >
            <div
              className="arrows left"
            >
              <div
                className="arrow previous"
                id="previous"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              />
            </div>
            <div
              className="arrows right"
            >
              <div
                className="arrow next"
                id="next"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              />
            </div>
            <div
              className="whatsHappen-card"
            >
              <div
                className="wh-normal-view"
              >
                <WhatsHappen
                  blog={Object {}}
                  count={0}
                />
                <WhatsHappen
                  blog={Object {}}
                  count={1}
                />
              </div>
              <div
                className="wh-responsive-view"
              >
                <WhatsHappen
                  blog={Object {}}
                  count="resp"
                />
              </div>
            </div>
          </div>,
          <div
            className="clear"
          />,
        ],
        "className": "wraper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Whats been happening
              </h2>,
              <span>
                What its like to be a part of our school community.
              </span>,
            ],
            "className": "des",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Whats been happening",
              },
              "ref": null,
              "rendered": "Whats been happening",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "What its like to be a part of our school community.",
              },
              "ref": null,
              "rendered": "What its like to be a part of our school community.",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="arrows left"
              >
                <div
                  className="arrow previous"
                  id="previous"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                />
              </div>,
              <div
                className="arrows right"
              >
                <div
                  className="arrow next"
                  id="next"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                />
              </div>,
              <div
                className="whatsHappen-card"
              >
                <div
                  className="wh-normal-view"
                >
                  <WhatsHappen
                    blog={Object {}}
                    count={0}
                  />
                  <WhatsHappen
                    blog={Object {}}
                    count={1}
                  />
                </div>
                <div
                  className="wh-responsive-view"
                >
                  <WhatsHappen
                    blog={Object {}}
                    count="resp"
                  />
                </div>
              </div>,
            ],
            "className": "slide-show",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="arrow previous"
                  id="previous"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                />,
                "className": "arrows left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow previous",
                  "id": "previous",
                  "onClick": [Function],
                  "onKeyPress": [Function],
                  "role": "button",
                  "tabIndex": "0",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="arrow next"
                  id="next"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                />,
                "className": "arrows right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow next",
                  "id": "next",
                  "onClick": [Function],
                  "onKeyPress": [Function],
                  "role": "button",
                  "tabIndex": "0",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="wh-normal-view"
                  >
                    <WhatsHappen
                      blog={Object {}}
                      count={0}
                    />
                    <WhatsHappen
                      blog={Object {}}
                      count={1}
                    />
                  </div>,
                  <div
                    className="wh-responsive-view"
                  >
                    <WhatsHappen
                      blog={Object {}}
                      count="resp"
                    />
                  </div>,
                ],
                "className": "whatsHappen-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <WhatsHappen
                        blog={Object {}}
                        count={0}
                      />,
                      <WhatsHappen
                        blog={Object {}}
                        count={1}
                      />,
                    ],
                    "className": "wh-normal-view",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "0.5167054428344668",
                      "nodeType": "function",
                      "props": Object {
                        "blog": Object {},
                        "count": 0,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": "0.6160745164367638",
                      "nodeType": "function",
                      "props": Object {
                        "blog": Object {},
                        "count": 1,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <WhatsHappen
                        blog={Object {}}
                        count="resp"
                      />,
                    ],
                    "className": "wh-responsive-view",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "0.44108523793400845",
                      "nodeType": "function",
                      "props": Object {
                        "blog": Object {},
                        "count": "resp",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wraper"
        >
          <div
            className="des"
          >
            <h2>
              Whats been happening
            </h2>
            <span>
              What its like to be a part of our school community.
            </span>
          </div>
          <div
            className="slide-show"
          >
            <div
              className="arrows left"
            >
              <div
                className="arrow previous"
                id="previous"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              />
            </div>
            <div
              className="arrows right"
            >
              <div
                className="arrow next"
                id="next"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              />
            </div>
            <div
              className="whatsHappen-card"
            >
              <div
                className="wh-normal-view"
              >
                <WhatsHappen
                  blog={Object {}}
                  count={0}
                />
                <WhatsHappen
                  blog={Object {}}
                  count={1}
                />
              </div>
              <div
                className="wh-responsive-view"
              >
                <WhatsHappen
                  blog={Object {}}
                  count="resp"
                />
              </div>
            </div>
          </div>
          <div
            className="clear"
          />
        </div>,
        "id": "sec-2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="des"
            >
              <h2>
                Whats been happening
              </h2>
              <span>
                What its like to be a part of our school community.
              </span>
            </div>,
            <div
              className="slide-show"
            >
              <div
                className="arrows left"
              >
                <div
                  className="arrow previous"
                  id="previous"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                />
              </div>
              <div
                className="arrows right"
              >
                <div
                  className="arrow next"
                  id="next"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                />
              </div>
              <div
                className="whatsHappen-card"
              >
                <div
                  className="wh-normal-view"
                >
                  <WhatsHappen
                    blog={Object {}}
                    count={0}
                  />
                  <WhatsHappen
                    blog={Object {}}
                    count={1}
                  />
                </div>
                <div
                  className="wh-responsive-view"
                >
                  <WhatsHappen
                    blog={Object {}}
                    count="resp"
                  />
                </div>
              </div>
            </div>,
            <div
              className="clear"
            />,
          ],
          "className": "wraper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Whats been happening
                </h2>,
                <span>
                  What its like to be a part of our school community.
                </span>,
              ],
              "className": "des",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Whats been happening",
                },
                "ref": null,
                "rendered": "Whats been happening",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "What its like to be a part of our school community.",
                },
                "ref": null,
                "rendered": "What its like to be a part of our school community.",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="arrows left"
                >
                  <div
                    className="arrow previous"
                    id="previous"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex="0"
                  />
                </div>,
                <div
                  className="arrows right"
                >
                  <div
                    className="arrow next"
                    id="next"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex="0"
                  />
                </div>,
                <div
                  className="whatsHappen-card"
                >
                  <div
                    className="wh-normal-view"
                  >
                    <WhatsHappen
                      blog={Object {}}
                      count={0}
                    />
                    <WhatsHappen
                      blog={Object {}}
                      count={1}
                    />
                  </div>
                  <div
                    className="wh-responsive-view"
                  >
                    <WhatsHappen
                      blog={Object {}}
                      count="resp"
                    />
                  </div>
                </div>,
              ],
              "className": "slide-show",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="arrow previous"
                    id="previous"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex="0"
                  />,
                  "className": "arrows left",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "arrow previous",
                    "id": "previous",
                    "onClick": [Function],
                    "onKeyPress": [Function],
                    "role": "button",
                    "tabIndex": "0",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="arrow next"
                    id="next"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="button"
                    tabIndex="0"
                  />,
                  "className": "arrows right",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "arrow next",
                    "id": "next",
                    "onClick": [Function],
                    "onKeyPress": [Function],
                    "role": "button",
                    "tabIndex": "0",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="wh-normal-view"
                    >
                      <WhatsHappen
                        blog={Object {}}
                        count={0}
                      />
                      <WhatsHappen
                        blog={Object {}}
                        count={1}
                      />
                    </div>,
                    <div
                      className="wh-responsive-view"
                    >
                      <WhatsHappen
                        blog={Object {}}
                        count="resp"
                      />
                    </div>,
                  ],
                  "className": "whatsHappen-card",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <WhatsHappen
                          blog={Object {}}
                          count={0}
                        />,
                        <WhatsHappen
                          blog={Object {}}
                          count={1}
                        />,
                      ],
                      "className": "wh-normal-view",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": "0.5167054428344668",
                        "nodeType": "function",
                        "props": Object {
                          "blog": Object {},
                          "count": 0,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": "0.6160745164367638",
                        "nodeType": "function",
                        "props": Object {
                          "blog": Object {},
                          "count": 1,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <WhatsHappen
                          blog={Object {}}
                          count="resp"
                        />,
                      ],
                      "className": "wh-responsive-view",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": "0.44108523793400845",
                        "nodeType": "function",
                        "props": Object {
                          "blog": Object {},
                          "count": "resp",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
